src/arrayMethodJoin.js

```diff
@@ -4,9 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (separator === null) {
+      for (let i = 0; i < this.length; i++) {
+        result += (i !== this.length - 1)
+          ? this[i] + String(separator) : this[i];
+      }
+    } else {
+      for (let i = 0; i < this.length; i++) {
```

Comment:
The checking of this[i] against null and undefined is not necessary as it's already done implicitly during string concatenation.